### DISCUSSION TITLE
Can Copier have conditional includes?

### DISCUSSION THREAD

**Discussion by kabdolha-amazon** [copier-org/copier#1070] [Q&A]
Hello, I'm interested in adopting copier for my team's scaffolding strategy. However, we want to have a high level template that you can choose different sub-templates depending on the project.

For example: If I have to choose Terraform or CDK, they will have different "Questionnaires" depending on the template. As far as I can tell, currently include is not conditional. How would I achieve this?

**Comment by sisp** [comment]
Would there be much overlap in the sub-templates, or would they be quite independent? I imagine that code, tests, CI config, etc. for Terraform looks quite different than for CDK. In that case, why not have two templates, one for each tool?

Copier assumes that there is only one template per repo. This makes sense in most cases.

Your question is related to #855. There, a [new approach](https://github.com/copier-org/copier/discussions/855#discussioncomment-5443008) was suggested a few days ago that uses `when`.
